### PR TITLE
Improving java docs by providing additional info and small usage samples

### PR DESCRIPTION
**Pull request by tangerinous**
# Description
The PR adds additional information to the java docs according to the [issue ](https://github.com/Azure/azure-sdk-for-java/issues/19815)

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by tangerinous** [comment]
Created by mistake

**Comment by tangerinous** [comment]
Sorry for my hesitation, this is part of the javadoc add-on, I will make similar edits for the rest of the methods based on the feedback results.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java [hunk 1]
```diff
@@ -1247,10 +1247,24 @@ public <TContext> Flux<CosmosBulkOperationResponse<TContext>> executeBulkOperati
     }
 
     /**
-     * Reads an item.
+     * Reads an item by itemId.
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java:1250]:
```suggestion
     * Reads an item by Id and partitionKey
```

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java [hunk 2]
```diff
@@ -1263,15 +1277,28 @@ public <T> Mono<CosmosItemResponse<T>> readItem(String itemId, PartitionKey part
     }
 
     /**
-     * Reads an item using a configured {@link CosmosItemRequestOptions}.
+     * Reads an item by itemId using a configured {@link CosmosItemRequestOptions}.
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java:1280]:
```suggestion
     * Reads an item by Id and partitionKey using a configured {@link CosmosItemRequestOptions}.
```

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
Thanks @tangerinous  for the changes. The Java doc description looks good, however, we should not put the code snippet directly, instead introduce new code snippets in the code snippets package and reference them here. 

An example is this - https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java#L2274

And code snippet for this example is here: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos/ThroughputControlCodeSnippet.java

You should add more code snippets here and reference them in the java docs. This will make things much easier, thanks!

https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java (hunk 2 above)
Comment (tangerinous) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java:1281]: Thanks @kushagraThapar for your feedback, I've removed unnecessary code examples from docs and added them to the samples class. 
Pushed new changes. Please tell me what else needs to be fixed

File: sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos/ReadmeSamples.java [hunk 3]
```diff
@@ -123,6 +126,20 @@ public void crudOperationOnItems() {
         // END: readme-sample-crudOperationOnItems
     }
 
+    public void readItem() {
+        // BEGIN: com.azure.cosmos.CosmosAsyncContainer.readItem
+        // Read an item
+        cosmosAsyncContainer.readItem(itemId, new PartitionKey(itemId), Passenger.class)
```
[the remark this answers is not in the document]
Comment (tangerinous) [reply sdk/cosmos/azure-cosmos/src/samples/java/com/azure/cosmos/ReadmeSamples.java]: done, but I hide Passenger object initialization, is it ok?

**Review by kushagraThapar — APPROVED** [review]
Yes, this looks good to me, thanks @tangerinous

**Pull request merged**